tests/basic/storage_pool_test.go

```diff
@@ -9761,3 +9761,190 @@ var _ = Describe("{AddDriveMetadataPool}", func() {
 		AfterEachTest(contexts)
 	})
 })
+
+var _ = Describe("{AddDriveBeyondMaxSupported}", func() {
+
+	/*
+		Add Drive using legacy add drive feature
+		test max drive per pool
+		test max pool per node
+		test max drives per node
+	*/
+	var testrailID = 19170501
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/tests/view/19170501
+	var runID int
+	JustBeforeEach(func() {
+		StartTorpedoTest("AddDriveBeyondMaxSupported", "Initiate pool Cloud add-drive beyond permitted and expect error", nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+	var contexts []*scheduler.Context
+
+	stepLog := "Test max limit of pool,node and max pool in a cluster"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("pooladddrivebeyondmax-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+		stNodes := node.GetStorageNodes()
+		if len(stNodes) == 0 {
+			dash.VerifyFatal(len(stNodes) > 0, true, "Storage nodes found?")
+		}
+		selectedNode := stNodes[0]
+		if selectedNode.Id == "" {
+			err := errors.New("failed to get node id")
+			log.FailOnError(err, "failed to get node id")
+		}
+
+		Step("Test max drives per pool", func() {
+			var selectedPool *api.StoragePool
+			var maxDrivesAllowed int
+			selectedPool, err := GetPoolWithIOsInGivenNode(selectedNode, contexts)
+			log.FailOnError(err, "error in getting pool with IO with error %s", err)
+			if selectedPool != nil {
+				drvMap, err := Inst().V.GetPoolDrives(&selectedNode)
+				log.FailOnError(err, "error getting pool drives from node [%s]", selectedNode.Name)
+				drvs := drvMap[fmt.Sprintf("%d", selectedPool.ID)]
+				maxDrivesAllowed = POOL_MAX_CLOUD_DRIVES - len(drvs)
+				fmt.Printf("maxDrivesAllowed %v", maxDrivesAllowed)
+			} else {
+				log.FailOnError(err, "error getting pool from node [%s]", selectedNode.Name)
+			}
+			for i := 1; i <= maxDrivesAllowed+1; i++ {
+				fmt.Printf("i %v", i)
+				drvSize, err := getPoolDiskSize(selectedPool)
+				log.FailOnError(err, "fialed while getting pool size")
+				driveSize := drvSize * uint64(i)
```

Comment:
do we need driveSize := drvSize * uint64(i) ? 

say maxDiskAllowed is 6 , and already a pool is with 3 disks , in this case calculation might go wrong , 
please recheck the logic here

Reply:
logic is working fine

Reply:
logic is correct

Reply:
made the changes to take different in case node limit is exceeded.Please find the pass logs after review comments are addressed